kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIUtil.java

```diff
@@ -0,0 +1,119 @@
+package io.elimu.serviceapi.config;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.elimu.a2d2.genericmodel.ServiceRequest;
+import io.elimu.a2d2.oauth.BodyBuilder;
+import io.elimu.a2d2.oauth.OAuthUtils;
+
+public class ConfigAPIUtil {
+
+	private static final Logger LOG = LoggerFactory.getLogger(ConfigAPIUtil.class);
+
+	private static final String AUTH_HEADER = "Authorization";
+	private static final String BEARER_PREFIX = "Bearer ";
+	
+	private final static Map<String, CachedResult> CACHE = new HashMap<>();
+	
+	public Map<String, Object> getConfig(ServiceRequest request, String environment, String client, String appName) {
+		Map<String, Object> retval = new HashMap<>();
+		String configApiUrl = System.getProperty("configApiUrl");
+		if (configApiUrl == null) {
+			return retval;
+		}
+		String token = getOrGenerateToken(request);
+		if (token == null) {
+			return retval;
+		}
+		String url = configApiUrl + "/" + client + "/" + environment + "/" + appName;
+		CachedResult result = CACHE.get(url);
+		if (result != null && !result.isOld()) {
+			retval.putAll(result.getVariables());
+		} else {
+			try (CloseableHttpClient httpclient = HttpClientBuilder.create().build()) {
+				HttpGet get = new HttpGet(url);
+				get.addHeader(AUTH_HEADER, BEARER_PREFIX + token);
+				HttpResponse response = httpclient.execute(get);
+				if (response.getStatusLine().getStatusCode() == 200) {
+					JsonNode data = new ObjectMapper().readTree(response.getEntity().getContent());
+					for(Iterator<String> iter = data.fieldNames(); iter.hasNext(); ) {
+						String key = iter.next();
+						JsonNode node = data.get(key);
+						if (node.isBoolean()) {
+							retval.put(key, node.asBoolean());
+						} else if (node.isInt()) {
+							retval.put(key, node.asInt());
+						} else if (node.isNumber()) {
+							retval.put(key, node.asDouble());
+						} else if (node.isTextual()) {
+							retval.put(key, node.asText());
+						} else if (node.isNull()) {
+							retval.put(key, null);
+						} else {
+							LOG.warn("Invalid value type for JSON property: " + node.getNodeType().name());
+						}
+					}
+				}
+				CACHE.put(url, new CachedResult(retval));
+			} catch (Exception e) {
+				LOG.warn("Error invoking config-api", e);
+			}
+		}
+		LOG.info("ConfigAPIUtil fetched {} variable values", CACHE.get(url).getVariables().size());
+		return retval;
+	}
+
+	private String getOrGenerateToken(ServiceRequest request) {
+		if (request != null && request.getHeader(AUTH_HEADER) != null && request.getHeader(AUTH_HEADER).startsWith(BEARER_PREFIX)) {
```

Comment:
@marianbuenosayres this if condition will alway's get executed, because, omnibus always going to expect `authorisation` header. how the below to if condition code will execute?

Reply:
@ddjain although sapphire will now call all omnibus services using OAuth, there is the possibility in the code to use x-api-key still. In those cases, we would invoke the code for generating the token